spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/orm/jpa/AbstractJpaAutoConfigurationTests.java

```diff
@@ -77,10 +80,40 @@ public void close() {
 	public void testNoDataSource() throws Exception {
 		this.context.register(PropertyPlaceholderAutoConfiguration.class,
 				getAutoConfigureClass());
-		this.expected.expect(BeanCreationException.class);
-		this.expected.expectMessage("No qualifying bean");
-		this.expected.expectMessage("DataSource");
 		this.context.refresh();
+		assertThat(this.context.getBeansOfType(JpaProperties.class).size()).isEqualTo(1);
```

Comment:
No. I am not sure I understand what triggers this change but previoulsy if you had no data source at all you'd get a `BeanCreationException`. Now you don't get any exception. That's the thing we're discussing there.